src/ui.py

```diff
@@ -12,22 +12,100 @@ def displayPlayerBoard(boards, player, hit):
         clear()
         input("PLAYER 2, look away. PLAYER 1, press enter when ready")
         clear()
+        displayFleetLife(PLAYER_1, boards)
         displayBoard(boards[PLAYER_1]["opponent"], "PLAYER 1'S TURN")
     elif player == PLAYER_2 and not hit:
         clear()
         input("PLAYER 1, look away. PLAYER 2, press enter when ready")
         clear()
+        displayFleetLife(PLAYER_2, boards)
         displayBoard(boards[PLAYER_2]["opponent"], "PLAYER 2'S TURN")
 
+def displayBoard(board, title="BOARD"):
+    print(title)
+    print(' ', end='')
+
+    # print header
+    for x in range(len(board)):
+        print(" " + str(x+1) + " ", end='')
+
+    # ascii codes for displaying grid
+    # upperleftcorner = u'\u2554'  
+    # downT = u'\u2569'
+    # T = u'\u2566'
+    # F = u'\u2560'
+    # line = u'\u2550'
+    # cross = u'\u256c'
+
+    print()
+
+    for x in range(len(board)):
+        # display letters using their unicode (starting form A)
+        line = chr(x + UNICODE_FOR_A_CHAR)
+
+        for i in range(len(board)):
+            if board[x][i] == SQUARE_SHIP:
+                line += " S "
+            elif board[x][i] == SQUARE_FAILED_SHOT:
+                line += " O "
+            elif board[x][i] == SQUARE_SUCCESS_SHOT:
+                line += " X "
+            else:
+                line += "   "
+
+        print(line)
+
 def displayShootResult(hit, boards, currentPlayer):
     clear()
 
     if (currentPlayer == PLAYER_1):
+        displayFleetLife(PLAYER_1, boards)
         displayBoard(boards[PLAYER_1]["opponent"])
     else:
+        displayFleetLife(PLAYER_2, boards)
         displayBoard(boards[PLAYER_2]["opponent"])
 
     if hit:
-        print("Well done, you blew up some ship. Play again!")
+        if not gameIsWon(boards):
+            print("Well done, you blew up some ship. Play again!")
     else:
         input("Too bad, you hit the water. Press enter to continue.")
+
+def displayFleetLife(player, boards):
+    print('YOUR FLEET LIFE: ' + str(boards[player]["life"]))
+
+def displayWinner(boards, player):
+    print("You blew up your oppponent's fleet! Congratulations " + getPlayerName(player) + "!")
+
+def getPlayerName(player):
+    name = 'PLAYER'
+
+    if player == PLAYER_1:
+        name = 'PLAYER 1'
+    else:
+        name = 'PLAYER 2'
+
+    return name
```

Comment:
```suggestion
def getPlayerName(player):
    if player == PLAYER_1:
        return 'PLAYER 1'
    else:
        return 'PLAYER 2'
```

You don't need a variable.

Reply:
I prefer to have only one return at the end of a function. That's why it is more explicit. What do you think ?

Reply:
It avoids a variable. But it's not a big deal :).